qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -110,32 +111,43 @@ def __init__(self, data, validate=True):
         """Initialize an operator object."""
 
         # Initialize from another Clifford by sharing the underlying
-        # StabilizerTable
+        # PauliList
         if isinstance(data, Clifford):
-            self._table = data._table
+            self._paulis = data._paulis
 
         # Initialize from ScalarOp as N-qubit identity discarding any global phase
         elif isinstance(data, ScalarOp):
             if not data.num_qubits or not data.is_unitary():
                 raise QiskitError("Can only initialize from N-qubit identity ScalarOp.")
-            self._table = StabilizerTable(np.eye(2 * data.num_qubits, dtype=bool))
+            array = np.eye(2 * data.num_qubits, dtype=bool)
+            x = array[:, 0 : data.num_qubits]
+            z = array[:, data.num_qubits : 2 * data.num_qubits]
+            self._paulis = PauliList.from_symplectic(z, x)
 
         # Initialize from a QuantumCircuit or Instruction object
         elif isinstance(data, (QuantumCircuit, Instruction)):
-            self._table = Clifford.from_circuit(data)._table
+            self._paulis = Clifford.from_circuit(data)._paulis
 
-        # Initialize StabilizerTable directly from the data
+        # Initialize PauliList directly from the data
         else:
-            self._table = StabilizerTable(data)
+            if isinstance(data, (list, np.ndarray)) and np.asarray(data, dtype=bool).ndim == 2:
+                data_array = np.asarray(data, dtype=bool)
+                num_qubits = data_array.shape[1] // 2
+                self._paulis = PauliList.from_symplectic(
+                    z=data_array[:, num_qubits : 2 * num_qubits],
+                    x=data_array[:, 0:num_qubits],
+                )
+            else:
+                self._paulis = PauliList(data)
```

Comment:
There may be a different case, but I can't come up with it. The test cases all pass, and since Clifford has an even number (stabilizer and destabilizer) of Paulis, it cannot possibly be given in one dimension, and I think it will fail in the validation below.